### PR TITLE
dockerize vrf and rr services for local setup

### PR DESCRIPTION
**Pull request by Intizar-T**
# Description

This PR is aimed to make setting up vrf and rr services locally easier and automate as much as possible. Running `docker-compose.local-core.yaml` will do the following (also explained in `dockerfiles/local-vrf-rr/README.MD`):
- api, postgres, redis, and json-rpc services will start as separate docker containers
- postgres will get populated with necessary data:
  - chains
  - services
  - vrf keys in case if the service is vrf
  - listener (after contracts are deployed)
  - reporter (after contracts are deployed)
- migration files in `contracts/v0.1/migration/` get updated with generated keys and other values
- relevant coordinator and prepayment contracts get deployed

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Checklist before requesting a review

- [x] I have performed a self-review of my code.
- [ ] If it is a core feature, I have added thorough tests.

## Deployment

- [ ] Should publish npm package
- [ ] Should publish Docker image

### CODE REVIEW

File: dockerfiles/local-vrf-rr/scripts/update-hardhat-network.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+
+const dir = "contracts/v0.1/hardhat.config.cjs";
+let fileContent = fs.readFileSync(dir, "utf8");
+const pattern = /(localhost: {\n\s*gasPrice: 250_000_000_000\n\s*},)/g;
+const replacement = `localhost: {
+    gasPrice: 250_000_000_000,
+    url: 'http://json-rpc:8545'
+  },`;
+fileContent = fileContent.replace(pattern, replacement);
+fs.writeFileSync(dir, fileContent);
```
[the remark this answers is not in the document]
Comment (Intizar-T) [reply dockerfiles/local-vrf-rr/scripts/update-hardhat-network.js:12]: the file being read is a js file. I don't think it's possible to import it, update its contents, and save back as json. The saved file should still be able to run as a js script

File: dockerfiles/local-vrf-rr/scripts/setup-vrf.sh [hunk 2]
```diff
@@ -0,0 +1,30 @@
+#!/bin/bash
+
+cd cli || exit
+
+yarn cli chain insert --name localhost
+yarn cli service insert --name VRF
+yarn cli service insert --name DATA_FEED
```
Comment (martinkersner) [inline dockerfiles/local-vrf-rr/scripts/setup-vrf.sh:7]:
```suggestion
```

**Review by martinkersner — APPROVED** [review]
LGTM! Finally, we have VRF and Request-Response! 🔥

**Comment by nick-bisonai** [comment]
I tried to run rr but kept facing following error from `core-cli-contracts` container
```
api-1                 | 2024/05/30 14:37:53 | 500 | 172.21.0.6 | POST | /api/v1/listener | ERROR: value too long for type character varying(42) (SQLSTATE 22001)
core-cli-contracts-1  | Listener was not inserted. Reason:
core-cli-contracts-1  | undefined
core-cli-contracts-1  | Done in 0.91s.
core-cli-contracts-1  | yarn run v1.22.19
core-cli-contracts-1  | $ node --no-warnings --import=specifier-resolution-node/register --experimental-json-modules dist/index.js reporter insert --chain localhost --service REQUEST_RESPONSE --address 0xf39Fd6e51aad88F6F4ce6aB8827279cffFb92266 --privateKey 0xac0974bec39a17e36ba4a6b4d238ff944bacb478cbed5efcae784d7bf4f2ff80 --oracleAddress '
core-cli-contracts-1  |
core-cli-contracts-1  |
core-cli-contracts-1  |
core-cli-contracts-1  |
core-cli-contracts-1  |
core-cli-contracts-1  |
core-cli-contracts-1  | 0xe7f1725E7734CE288F8367e1Bb143E90bb3F0512
core-cli-contracts-1  | 0x8464135c8F25Da09e49BC8782676a84730C318bC'
```
it failed to insert both listener and reporter, am I missing something 🤔

**Comment by nick-bisonai** [comment]
and this kind of error cam out when trying to generate new prepayment account ID from consumer repo
```
2024-05-30 23:47:08 eth_sendTransaction
2024-05-30 23:47:08   Contract call:       Prepayment#createAccount
2024-05-30 23:47:08   Transaction:         0xb4ecae1bcf2704ce9e0b6eb9ccf1d0f0e9d5546bff7976a2014a7026acec230e
2024-05-30 23:47:08   From:                0xf39fd6e51aad88f6f4ce6ab8827279cfffb92266
2024-05-30 23:47:08   To:                  0x5fbdb2315678afecb367f032d93f642f64180aa3
2024-05-30 23:47:08   Value:               0 ETH
2024-05-30 23:47:08   Gas used:            415675 of 421272
2024-05-30 23:47:08   Block #10:           0x18e5978c6264a6d1ab36f144862856c10853117be4b96bc2e569449873bab792
2024-05-30 23:47:08 
2024-05-30 23:47:08   Error: Transaction reverted without a reason
```
Am I missing any settings? 🤔

**Review by nick-bisonai — APPROVED** [review]
LGTM!

**Pull request merged**